File: 884.uncommon-words-from-two-sentences.go
package main

import (
	"strings"
)

/*
 * @lc app=leetcode id=884 lang=golang
 *
 * [884] Uncommon Words from Two Sentences
 */

// @lc code=start
func uncommonFromSentences(s1 string, s2 string) []string {
	h := make(map[string]int)
	var result []string
	hoge := append(strings.Split(s1, " "), strings.Split(s2, " ")...)
	for _, s := range hoge {
		h[s]++
	}

	for k, v := range h {
		if v == 1 {
			result = append(result, k)
		}
	}
	return result
}

// @lc code=end

File: 884.uncommon-words-from-two-sentences.rb
#
# @lc app=leetcode id=884 lang=ruby
#
# [884] Uncommon Words from Two Sentences
#

# @lc code=start
# @param {String} s1
# @param {String} s2
# @return {String[]}
def uncommon_from_sentences(s1, s2)
  h = {}
  result = []
  (s1.split + s2.split).each { |c| h[c] ? h[c] += 1 : h[c] = 1 }

  h.each do |k, v|
    result << k if v == 1
  end
  result
end
# @lc code=end

